dashboard/modules/serve/serve_agent.py

```diff
@@ -251,51 +251,44 @@ async def put_all_applications(self, req: Request) -> Response:
         from ray._private.usage.usage_lib import TagKey, record_extra_usage_tag
 
         try:
-            config = ServeDeploySchema.parse_obj(await req.json())
+            config: ServeDeploySchema = ServeDeploySchema.parse_obj(await req.json())
         except ValidationError as e:
             return Response(
                 status=400,
                 text=repr(e),
             )
 
+        config_http_options = {
```

Comment:
Maybe just do `config_http_options = config.http_options.dict()` so everything will be included

Reply:
Yes this will prevent similar issues from happening in the future

Reply:
Good idea, that makes the code a lot simpler too!